app/helpers/quadicon_helper.rb

```diff
@@ -70,8 +70,6 @@ def self.settings_key(klass, layout)
       when "ems_physical_infra", "ems_cloud", "ems_network", "ems_container"
         layout.to_sym
       end
-    elsif klass.base_model.name != klass.name.demodulize
```

Comment:
No. This will break Vm for example :).

Reply:
Ok, I put back the demodulize and added a condition for "physical_switch".

Reply:
Yes, it is safe to remove. It was introduced by me at https://github.com/ManageIQ/manageiq-ui-classic/pull/3699/files#diff-6525cb9600835fcc2ba062fde39a2292R73

Reply:
@himdel @saulotoledo I removed both demodulize and the `elsif` for `physical_switch`.

Reply:
@himdel this actually fixes VMs :wink:

Reply:
But as I stated it [here](https://github.com/ManageIQ/manageiq/pull/17431#issuecomment-391591783) this might not be *The Right Way&trade;* to fix this...